Formula/php.rb

```diff
@@ -0,0 +1,271 @@
+      s.change_make_var! "EXTRA_LIBS", "\\1 -lstdc++"
+    end
+
+    system "make"
+    ENV.deparallelize
+    system "make", "install"
+
+    ln_s "phar.phar", bin+"phar", :force => true
+
+    # Install new php.ini unless one exists
+    config_path.install "./php.ini-development" => "php.ini" unless File.exist? config_path+"php.ini"
+
+    chmod 0755, "sapi/fpm/init.d.php-fpm"
+    sbin.install "sapi/fpm/init.d.php-fpm" => "php#{version.to_s[0..2].delete(".")}-fpm"
+
+    if File.exist?("sapi/cgi/fpm/php-fpm")
+      chmod 0755, "sapi/cgi/fpm/php-fpm"
+      sbin.install "sapi/cgi/fpm/php-fpm" => "php#{version.to_s[0..2].delete(".")}-fpm"
+    end
+
+    if !File.exist?(config_path+"php-fpm.d/www.conf") && File.exist?(config_path+"php-fpm.d/www.conf.default")
+      mv(config_path+"php-fpm.d/www.conf.default", config_path+"php-fpm.d/www.conf")
+    end
+
+    unless File.exist?(config_path+"php-fpm.conf")
+      if File.exist?("sapi/fpm/php-fpm.conf")
+        config_path.install "sapi/fpm/php-fpm.conf"
+      end
+
+      if File.exist?("sapi/cgi/fpm/php-fpm.conf")
+        config_path.install "sapi/cgi/fpm/php-fpm.conf"
+      end
+
+      inreplace config_path+"php-fpm.conf" do |s|
+        s.sub!(/^;?daemonize\s*=.+$/, "daemonize = no")
+      end
+    end
+  end
+
+  def caveats
+    s = []
+
+    s << <<-EOS.undent
+      To enable PHP in Apache add the following to httpd.conf and restart Apache:
+          LoadModule php7_module #{HOMEBREW_PREFIX}/opt/php#{version.to_s[0..2].delete(".")}/libexec/apache2/libphp7.so
+
+          <FilesMatch \.php$>
+              SetHandler application/x-httpd-php
+          </FilesMatch>
+
+      Finally, check DirectoryIndex includes index.php
+          DirectoryIndex index.php index.html
+    EOS
+
+    s << <<-EOS.undent
+      The php.ini file can be found in:
+          #{etc}/php/#{version.to_s.split(".")[0..1].join(".")}/php.ini
+    EOS
+
+    s.join "\n"
+  end
+
+  def post_install
+    (prefix/"var/log").mkpath
+    touch prefix/"var/log/php-fpm.log"
+
+    chmod 0755, lib/"php/.channels"
+    chmod 0755, lib/"php/.channels/.alias"
+    chmod 0644, (Dir.glob(lib/"php/.channels/**/*", File::FNM_DOTMATCH).reject { |a| a =~ /\/\.{1,2}$/ || File.directory?(a)})
+
+    %w[
+      php/.depdblock
+      php/.filemap
+      php/.depdb
+      php/.lock
+    ].each do |f|
+      chmod 0644, lib/f
+    end
+
+    php_ini = "#{etc}/php/#{version.to_s.split(".")[0..1].join(".")}"
+    system bin/"pear", "config-set", "php_ini", php_ini, "system"
+
+    %w[
+      ldap
+      mcrypt
+      tidy
+      imap
+      opcache
+    ].each do |e|
+      config_path = (etc/"php/#{version.to_s.split(".")[0..1].join(".")}/conf.d/ext-#{e}.ini")
+      extension_type = (e == "opcache") ? "zend_extension" : "extension"
+      if File.exist? config_path
+        inreplace config_path do |s|
+          s.gsub /#{extension_type}=.*$/, "#{extension_type}=#{Utils.popen_read("php-config --extension-dir").chomp}/#{e}.so"
```

Comment:
I don't think the `Utils.popen_read` is necessary. The extensions will be loaded from the configured `extension_dir` already. Having the full path in these config files make the configuration less portable, say when you migrate your configuration to the next major version or even when you upgrade the formula version. The default `extension_dir` will be the cellar path. `extension=ldap.so` should suffice.

This also brings up an interesting side-effect. Extensions that are then installed with `pecl` will be installed into the cellar. This means that they won't survive a formula patch version bump unless the full path to the `.so` file installed is in the `php.ini`. I'm not sure what the best solution for this would look like. Installing to a directory outside of the cellar would be helpful, but the config written would still need to be changed and determining a good directory that homebrew doesn't complain about is a challenge.

Reply:
Ah yes, that had eluded me. I will remove the popen part.

Your second comment had also struck me but I don't have a solution for that yet.

Reply:
PIP for pythons uses `Location: /usr/local/lib/python2.7/site-packages` perhaps that would be an option